svcs/amigo/amigo.go

```diff
@@ -925,3 +940,46 @@ func parseTemplates(givenTemplate string) (*template.Template, error) {
 	)
 	return tmpl, err
 }
+
+func checkTeamName(input string) error {
+	re := regexp.MustCompile("^[A-Za-z0-9]+$")
+
+	if input == "" {
+		return ErrTeamNameEmpty
+	}
+
+	if err := checkVarLength(input, TeamNameMaxLength); err != nil {
+		return ErrTeamNameToLarge
+	}
+
+	if !re.MatchString(input) {
+		return ErrTeamNameCharacters
+	}
+
+	return nil
+}
+
+func checkEmail(input string) error {
+	re := regexp.MustCompile("^[a-zA-Z0-9.!#$%&'*+^`{|}~]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$")
+
+	if input == "" {
+		return ErrEmailEmpty
+	}
+
+	if err := checkVarLength(input, emailMaxLength); err != nil {
+		return ErrEmailToLarge
+	}
+
+	if !re.MatchString(input) {
+		return ErrEmailCharacters
+	}
+
+	return nil
+}
+
+func checkVarLength(input string, max int) error {
+	if len(input) >= max {
+		return fmt.Errorf("Input string is not within the defined character limit")
```

Comment:
I think, this is forgotten to return `ErrEmailTooLarge` error here.

Reply:
This function is also used to check the length of the team name length